component/client/service/RoomService.tsx

```diff
@@ -24,13 +21,13 @@ class RoomService {
     });
   }
 
-  public pushMessage(message: string) {
-    if (!message) throw new Error(`message is ${message}`);
-    this.ref.push().set(message);
+  public pushMessage(data: T) {
+    if (!data) throw new Error(`data is ${data}`);
+    this.ref.push().set(data);
   }
 
   public close() {
-    this.ref.remove();
+    this.ref.off();
```

Comment:
remove() will remove all of the data, and whoever still on this page will see all messages suddenly disappearing. 
You just want to unlink the listener.